matrix/tasks/glitch/main.py

```diff
@@ -106,10 +108,20 @@ def default_resolver(model, kind, name):
         # Run the specified action on those units
         rule.log.info("GLITCHING {}: {}".format(actionf.__name__, objects))
 
+        errors = False
         try:
             await asyncio.wait_for(actionf(rule, model, objects, **action), 30)
         except asyncio.TimeoutError:
             rule.log.error("Timeout running {}".format(actionf.__name__))
+            errors = True
+        except (TypeError, KeyError, AttributeError, IndexError) as e:
+            rule.log.error(
+                "Exception while running {}: {} {}.".format(
+                    actionf.__name__, type(e), e))
+            errors = True
+        if errors and task.gating:
+            raise TestFailure(task, "Exceptions were raised during glitch run.")
```

Comment:
On the other hand, perhaps we should consider making task exceptions always gating, since it seems like they'd indicate a bug in Matrix.